### DISCUSSION TITLE
런타임시 카메라 설정

### DISCUSSION THREAD

**Discussion by koldshine** [naverz/zepeto-studio-kor#961] [5. Q&A]
안녕하세요
상황에 따라 기본3D뷰와 TPS뷰사이에서 전환하고 싶은데요
tpsview 예제에서 zoom min, max, offset 값을  세팅하던데 
이걸 런타임에서도 변경하는 방법이 있을까요?

**Comment by Zepeto-TechEvangelist** [comment]
안녕하세요. 말씀하신 제페토 카메라의 설정 값들은 현재는 초기화 타이밍에 결정되는 값이라 런타임 중 수정은 불가능 합니다.
커스텀 카메라를 생성해서 제어하시는 것이 좋을 것 같습니다.

**Reply by koldshine** [reply]
지금 막 찾아서 답글 달려고 했는데요
ZepetoPlayers.instance.cameraData로 실행중에도 위 값들 변경이 가능하네요. 
혹시 사용하면 안 되는건가요? 나중에 막힐 수 있는지 궁금합니다

**Reply by Zepeto-TechEvangelist** [reply]
ZepetoPlayers.instance.cameraData 는 ZepetoPlayers에서 관리하는 모든 ZepetoCamera에 대한 값에 영향을 미칠 수 있으므로 사용을 권장 드리지 않습니다.  
현재 런타임에 열려있지 않은 변수를 다음 업데이트에 추가로 열어드릴 예정입니다.
감사합니다.

**Reply by koldshine** [reply]
답변 감사합니다

**Comment by YSHFM** [comment]
OnPreCull 단계에서 따로 캠의 localTransform값을 조정한다면 offset또는 rotation값을 조정할 수 있을 것 같습니다.

![image](https://user-images.githubusercontent.com/106574541/177278694-38fffefa-337d-43c6-8e1c-61de99a9a0ef.png)

![image](https://user-images.githubusercontent.com/106574541/177278824-d1260beb-7035-448c-8b6b-3fd37a968def.png)

실제로 사용중인 코드 예시이며 이 코드를 이용해 sideview를 사용하고 있습니다.
transform값을 임의로 조정한다면 제어할 수 있지만 기본적인 렌더링 파이프라인 지식을 요구할 수 있습니다.
